### PR TITLE
Deeper measure total

### LINKED ISSUE #115: investigate and fix evmone overhead in docker

### PR DESCRIPTION
**Pull request by pdobacz**
Closes https://github.com/imapp-pl/gas-cost-estimator/issues/115

Related to https://github.com/imapp-pl/evmc/pull/4

DO NOT MERGE! reset wallclock branch to here and use in wallclock evmone.

The new addition is the [Move the measurement decorations deeper down the stack](https://github.com/imapp-pl/evmone/commit/e4223bbc351f062116154628f6bc5d2e08cb34ab), but pls review the whole thing.

### CODE REVIEW

File: lib/evmone/baseline.cpp [hunk 1]
```diff
@@ -271,12 +273,37 @@ evmc_result execute(const VM& vm, ExecutionState& state, const CodeAnalysis& ana
...
-    auto state =
-        std::make_unique<ExecutionState>(*msg, rev, *host, ctx, bytes_view{code, code_size});
-    return execute(*vm, *state, jumpdest_map);
+
+    evmc_result result;
+
+    // gas-cost-estimator decoration
+    //
+    for (unsigned int run_id = 0; run_id < sample_size; ++run_id) {
+        auto state =
```
Comment (chfast) [inline lib/evmone/baseline.cpp:285]: You should allocate it outside of the loop. Inside the loop use `state->reset(...)`.

File: lib/evmone/baseline.cpp [hunk 2]
```diff
@@ -271,12 +273,37 @@ evmc_result execute(const VM& vm, ExecutionState& state, const CodeAnalysis& ana
...
+            std::make_unique<ExecutionState>(*msg, rev, *host, ctx, bytes_view{code, code_size});
+        std::chrono::steady_clock::time_point start_time, end_time, end_time_timer;
+        start_time = std::chrono::steady_clock::now();
+
+        const auto jumpdest_map = analyze({code, code_size});
```
Comment (chfast) [inline lib/evmone/baseline.cpp:290]: The analysis is also constant per code so you can move outside of the loop.

File: lib/evmone/baseline.cpp (hunk 1 above)
Comment (pdobacz) [reply lib/evmone/baseline.cpp:285]: as discussed - it's better to `reset`, but not a big error to realloc. I'll make a follow-up issue out of this

File: lib/evmone/baseline.cpp (hunk 2 above)
Comment (pdobacz) [reply lib/evmone/baseline.cpp:290]: as discussed - we're keeping analysis inside the measurement to be more "fair"

**Comment by pdobacz** [comment]
fast-forwarded `wallclock` to this commit

**Pull request closed without merging**